best-practices/testing.md

```diff
@@ -1,15 +1,42 @@
 # Testing
 
+## Why?
+
 Testing helps show how the code was intended to be used, proves that it works for
 the important cases, and prevents future developers from introducing regressions.
 Having a good test suite enables us to refactor the code without worrying that
-we're going to break something.  Every feature addition should be submitted with
-at least one test.
+we're going to break something.
+
+## When?
+
+Every feature addition should be submitted with at least one (meaningful\*) test. Ideally,
```

Comment:
What's the `\*`? Markdown-splosion?

Reply:
just to get an actual asterisk instead of a bullet

Reply:
ah, sorry, i was thinking about the one at the end.  the lone one here might need no escape?